### PR TITLE
SISRP-20755 Integrate new Terms API for Fall 2016 and after

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.berkeley.edu/browse/SISRP-20755

As usual with these epic PRs, I've organized the commits for easier code review.

### CODE REVIEW

File: app/models/my_academics/academics_module.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def semester_info(term_key)
         termCode: term_cd,
         termYear: term_yr,
         timeBucket: time_bucket(term_yr, term_cd),
-        campusSolutionsTerm: Berkeley::Terms.campus_solutions?(term_yr, term_cd),
+        campusSolutionsTerm: !Berkeley::Terms.legacy?(term_yr, term_cd),
```
Comment (paulkerschen) [inline app/models/my_academics/academics_module.rb]: @raydavis Is there any false-versus-nil gotcha in making this change?

**Comment by paulkerschen** [comment]
I have read through, and I fear it makes sense, but @johncrossman may also wish to review.

File: app/models/my_academics/academics_module.rb (hunk 1 above)
Comment (raydavis) [inline app/models/my_academics/academics_module.rb]: Reasonable question, but no, not in this context.

**Comment by raydavis** [comment]
Annoyingly a stray bit of unused code slipped into my first PR, so I had to re-push.

**Comment by davidlimcheng** [comment]
Couldn't find anything questionable here, @raydavis.

**Comment by johncrossman** [comment]
@raydavis I'm reviewing now. Merge a comin'

**Comment by johncrossman** [comment]
Nice work, @raydavis  Well crafted. Allons-y!

**Pull request merged**